src/klein/app.py

```diff
@@ -329,12 +330,33 @@ def run(self, host, port, logFile=None):
 
         @param logFile: The file object to log to, by default C{sys.stdout}
         @type logFile: file object
+
+        @param proto: protocol string for endpoint, can be either "tcp" or "ssl"
+        @type proto: str
+
+        @param privateKey: filepath to private key for ssl
+        @type privateKey: str
+
+        @param certificate: filepath to certificate for ssl
+        @type certificate: str
+
+        @param dhParameters: filepath to dh parameters for ssl
+        @type dhParameters: str
         """
         if logFile is None:
             logFile = sys.stdout
 
         log.startLogging(logFile)
-        reactor.listenTCP(port, Site(self.resource()), interface=host)
+        server_string = "{}:{}:interface={}".format(proto, port, host)
+
+        if privateKey and certificate:
+            server_string += ":privateKey={}:certKey={}".format(privateKey,
+                                                                certificate)
+        if dhParameters:
+            server_string += ":dhParameters={}".format(dhParameters)
+
+        endpoint = endpoints.serverFromString(reactor, server_string)
+        endpoint.listen(Site(self.resource()))
```

Comment:
> Because it allows users to use the full grammar of endpoints

hmm yes this sounds convincing.

> So if we're going to use serverFromString, we should just let the user pass whatever endpoint string they want.

but this will create two different (possibly duplicated) forms of passing arguments to run(), one would accept host, port etc (as it does now), and other would accept server string specification, to pass string specification you'd have to call run like this `run(host, port, server_string='tcp:port:host")` 

Maybe it would be better to just add another method to app.Klein object that would take optional server string and it would run serverFromString on that? E.g. add method `run_from_string(server_string)`

> or otherwise use the TCP4ServerEndpoint directly in the way the old code used listenTCP.

given the amount of options in SSL it might be slightly inconvenient to pass all of them to run, run would have to take lots of optional keyword arguments defaulting to None. I think I would vote for passing string specification directly and doing serverFromString on that.